cirq/contrib/quirk/cells/arithmetic_cells.py

```diff
@@ -49,6 +49,13 @@ def __init__(self, identifier: str, target: Sequence['cirq.Qid'],
         self.target = target
         self.inputs = inputs
 
+        for input_register in self.inputs:
```

Comment:
Quirk input registers are technically allowed to overlap ( https://algassert.com/quirk#circuit={%22cols%22:[[%22X%22,1,1,%22X%22],[%22inputA3%22,1,%22inputB3%22,1,1,1,1,%22+=AB3%22]]} ), it just doesn't really come up because the program fights you on it.

I would split this out into a separate PR.

Reply:
In general an operation with overlapping inputs is well-defined. Though I can see why this would confuse the moment overlap check down the line.